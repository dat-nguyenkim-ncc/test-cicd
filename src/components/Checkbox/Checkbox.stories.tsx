import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import Checkbox, { CheckboxProps } from './Checkbox'
import { Section } from '../primitives'

export default {
  title: 'Form Elements/Checkbox',
  component: Checkbox,
} as Meta

const Template: Story<CheckboxProps> = args => (
  <Section sx={{ m: 6, mx: 'auto' }}>
    <Checkbox {...args} />
  </Section>
)

export const Default = Template.bind({})
Default.args = {} as CheckboxProps
