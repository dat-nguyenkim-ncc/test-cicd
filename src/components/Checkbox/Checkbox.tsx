import React, { MouseEvent } from 'react'
import { Box, Flex, Text } from 'theme-ui'
import { MapVariantToSize, Size, ViewInterface } from '../../types'
import { Icon } from '../'
// import { Paragraph } from '../primitives'
import { Palette } from '../../theme'

const sizes: MapVariantToSize = {
  big: 80,
  small: 24,
  tiny: 16,
  normal: 40,
}

export type CheckboxProps = ViewInterface<{
  checked?: boolean
  onPress?(e: MouseEvent): void
  label?: string
  square?: boolean
  disabled?: boolean
  size?: Size
  gap?: number
  prefix?: React.ReactElement
}>

const Checkbox = ({
  checked,
  onPress,
  label,
  square = false,
  disabled = false,
  sx,
  size,
  gap,
  prefix,
}: CheckboxProps) => {
  return (
    <Flex
      sx={{
        alignItems: 'center',
        cursor: onPress ? 'pointer' : 'default',
        opacity: disabled ? 0.5 : 1,
        pointerEvents: disabled ? 'none' : 'unset',
        ...sx,
      }}
      onClick={(e: MouseEvent<any>) => {
        if (!disabled) onPress && onPress(e)
      }}
    >
      <Box
        sx={{
          borderWidth: checked ? 0 : '1px',
          borderStyle: 'solid',
          borderColor: 'gray04',
          width: size ? sizes[size] : square ? sizes.tiny : sizes.small,
          height: size ? sizes[size] : square ? sizes.tiny : sizes.small,
          borderRadius: square ? '3px' : '100%',
          background: Palette.white,
        }}
        className={checked ? 'checkbox checked' : 'checkbox'}
      >
        <Icon
          icon={'tick'}
          background={'primary'}
          color="white"
          size={size || (square ? 'tiny' : 'small')}
          iconSize={size === 'tiny' ? 10 : undefined}
          sx={{
            borderRadius: square ? '3px' : '100%',
            opacity: checked ? 1 : 0,
          }}
        />
      </Box>
      {prefix}
      {label && (
        <Text
          as="p"
          sx={{
            pl: gap || (square ? 3 : 4),
            maxWidth: `calc(100% - ${square ? sizes.tiny : sizes.small}px)`,
            fontWeight: !square && checked ? 'bold' : 'body',
          }}
          variant="body"
        >
          {label}
        </Text>
      )}
    </Flex>
  )
}

export default Checkbox
