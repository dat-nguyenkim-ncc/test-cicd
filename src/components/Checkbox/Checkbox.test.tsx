import * as React from 'react'
import { shallow } from 'enzyme'
import Checkbox from './Checkbox'

describe('Checkbox', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<Checkbox checked />)
    expect(wrapper).toMatchSnapshot()
  })
})
