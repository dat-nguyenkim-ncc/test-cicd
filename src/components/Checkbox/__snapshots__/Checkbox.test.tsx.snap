// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox matches snapshot 1`] = `ShallowWrapper {}`;
